Implemented basic filter functionality

nomarchia opened the pull request:
1) Add AbstractSecurityFilter to extract same functionality;
2) Add Admin and User SecurityFilters extending the AbstractSecurityFilter;
3) Add filters mapping to the web.xml;

nomarchia committed:
Initial commit:
1) Add AbstractSecurityFilter to extract same functionality;
2) Add Admin and User SecurityFilters extending the AbstractSecurityFilter;
3) Add filters mapping to the web.xml;

Brutt commented on src/main/webapp/WEB-INF/web.xml:
```diff
@@ -71,4 +71,28 @@
+        <filter-name>UserRoleFilter</filter-name>
+        <url-pattern>/home</url-pattern>
+        <url-pattern></url-pattern>
+        <dispatcher>REQUEST</dispatcher>
```
REQUEST is the default

Brutt commented on src/main/webapp/WEB-INF/web.xml:
```diff
@@ -71,4 +71,28 @@
+        <filter-name>AdminRoleFilter</filter-name>
+        <url-pattern>/admin</url-pattern>
+        <url-pattern>/admin/users</url-pattern>
+        <url-pattern>/orders</url-pattern>
```
User also needs to have access to orders (his or her orders)

Brutt commented on src/main/java/com/photostudio/web/filter/AbstractSecurityFilter.java:
```diff
@@ -0,0 +1,65 @@
+        HttpServletRequest req = (HttpServletRequest) request;
+        HttpServletResponse resp = (HttpServletResponse) response;
+
+        Cookie[] cookies = req.getCookies();
```
we have util class CookieManager for this purpose

nomarchia committed:
b-19-add-filters-for-pages-by-role-nomar
1) Removed default <dispatcher> value from filter mapping;
2) Added /orders url for UserSecurityFilter too;
3) Renamed constants to the PROPER_CASE;
4) Used existing util class CookieManager instead of iterating cookies manually in filter;

Brutt commented on src/main/java/com/photostudio/web/filter/AbstractSecurityFilter.java:
```diff
@@ -0,0 +1,58 @@
+        //boolean isAuth = false;
+        //UserRole userRole;
+
+        String token = new CookieManager().getCookie(req,"token");
```
make new CookieManager() as private variale

Brutt commented on src/main/java/com/photostudio/web/filter/AbstractSecurityFilter.java:
```diff
@@ -0,0 +1,58 @@
+    public void doFilter(ServletRequest request, ServletResponse response, FilterChain chain) throws IOException, ServletException {
+        HttpServletRequest req = (HttpServletRequest) request;
+        //This variables would be needed to add redirection. See comment in the end of method
+        //HttpServletResponse resp = (HttpServletResponse) response;
```
remove commented code

Brutt commented on src/main/java/com/photostudio/web/filter/AbstractSecurityFilter.java:
```diff
@@ -0,0 +1,58 @@
+            }
+        }
+
+        //I can add a redirection here. For example if user tries to visit admin page,
```
Implement access denied page and redirect to it in suck cases

nomarchia committed:
b-19-add-filters-for-pages-by-role-nomar
1) Add AccessDeniedServlet;
2) Add access_denied .html + .css;
3) Add redirection in AbstractSecurityFilter;
4) Extract CookieManager to private variable in LoginServlet;
5) Filter mapping minor changes (unfinished);
